kasa/aestransport.py

```diff
@@ -102,31 +111,28 @@ def credentials_hash(self) -> str:
         """The hashed credentials used by the transport."""
         return base64.b64encode(json_dumps(self._login_params).encode()).decode()
 
-    def _get_login_params(self, credentials):
+    def _get_login_params(self, credentials: Credentials) -> Dict[str, str]:
         """Get the login parameters based on the login_version."""
         un, pw = self.hash_credentials(self._login_version == 2, credentials)
         password_field_name = "password2" if self._login_version == 2 else "password"
         return {password_field_name: pw, "username": un}
 
     @staticmethod
-    def hash_credentials(login_v2, credentials):
+    def hash_credentials(login_v2: bool, credentials: Credentials) -> Tuple[str, str]:
         """Hash the credentials."""
+        un = base64.b64encode(_sha1(credentials.username.encode()).encode()).decode()
         if login_v2:
-            un = base64.b64encode(
-                _sha1(credentials.username.encode()).encode()
-            ).decode()
             pw = base64.b64encode(
                 _sha1(credentials.password.encode()).encode()
             ).decode()
         else:
-            un = base64.b64encode(
-                _sha1(credentials.username.encode()).encode()
-            ).decode()
             pw = base64.b64encode(credentials.password.encode()).decode()
         return un, pw
 
-    def _handle_response_error_code(self, resp_dict: dict, msg: str):
-        error_code = SmartErrorCode(resp_dict.get("error_code"))  # type: ignore[arg-type]
+    def _handle_response_error_code(self, resp_dict: Any, msg: str) -> None:
+        error_code = SmartErrorCode(
+            (resp_dict or {}).get("error_code", SmartErrorCode.UNSPECIFIC_ERROR.value)
```

Comment:
This feels wrong., maybe the directory conversion should be done at the call-site, and this here would just do `resp_dict.get("error_code", SmartErrorCode.UNSPECIFIC_ERROR.value)`?

Reply:
I guess its possible that we won't get a dict back at all .. maybe some other error

Reply:
Probably makes sense just to revert the change for now since I don’t want to change the test in the same PR that I refactor to minimize regression risk

Reply:
reverted to master version in e88e7c3901fe799d189e2d4cb52af72380a69ea6